katana/utils.py

```diff
@@ -15,89 +16,69 @@ def parse_date(value):
-#     action += ' -ncpu 1 -s '
-#     # add each station location
-#     for lon, lat in zip(lons, lats):
-#         action += ' -lon {} {}'.format(lon, lat)
-
-#     print('\nTrying {}\n'.format(action))
-#     s = Popen(action, shell=True, stdout=PIPE, stderr=PIPE)
-#     idl = 0
-#     # array to store values
-#     val = np.ones(len(lats))*np.nan
-
-#     real_lon = []
-#     real_lat = []
-
-#     while True:
-
-#         line = s.stdout.readline().decode()
-#         eline = s.stderr.readline().decode()
-#         if not line:
-#             break
-
-#         if "FATAL" in eline:
-#             raise ValueError('Error in wgrib2: \n{}'.format(eline))
-
-#         # find the val, lon, lat
-#         # print(line)
-#         lines = line.split('::')[1].split(':')
-#         # print(lines)
-#         if len(lines) != len(stns):
-#             raise ValuesError('Not enough returns')
-
-#         for idl, ln in enumerate(lines):
-#             # get value from return string
-#             out = float(re.search('val=(.+?)$', ln).group(1))
-#             # get coordinate of pixel from return string
-#             real_lon.append(float(re.search('lon=(.+?),', ln).group(1)))
-#             real_lat.append(float(re.search('lat=(.+?),', ln).group(1)))
-
-#             # convert air temp to celcius
-#             if k == 'air_temp':
-#                 out -= 273.15
-#             # store value at staion index
-#             val[idl] = out
-
-#     # loop through each station and grab value at that station
-#     for idl, stn in enumerate(stns):
-#         df[stn] = [val[idl]]
-#         df_loc.loc[stn, 'latitude'] = real_lat[idl]
-#         df_loc.loc[stn, 'longitude'] = real_lon[idl]
-
-#     return df, df_loc
+
+def daterange(start_date, end_date, delta=timedelta(hours=1)):
+    """Create a `datetime` list between the start and end date
+    by a given timedelta.
+
+    Arguments:
+        start_date {datetime} -- start date
+        end_date {datetime} -- end date
+
+    Keyword Arguments:
+        delta {timedelta} -- timedelta for the step
+                    (default: {timedelta(hours=1)})
+
+    Returns:
+        [list] -- List of `datetime` objects
+    """
+
+    dr = []
+    while start_date <= end_date:
+        dr.append(start_date)
+        start_date += delta
+
+    return dr
+
+
+def daylist(start_date, end_date):
+    """Create a list of days between two dates
+
+    Arguments:
+        start_date {datetime.datetime} -- start date for list
+        end_date {datetime.datetime} -- end date for list
+
+    Returns:
+        list -- list of days between the provided dates
+    """
+
+    # get list of days to grab
+    dtt = end_date - start_date
+    ndays = int(dtt.days)
+    start_midnight = date(
+        start_date.year, start_date.month, start_date.day)
+    day_list = [start_midnight +
+                timedelta(days=x) for x in range(0, ndays+1)]
+
+    return day_list
+
+
+class CheckRawString(CheckType):
```

Comment:
At some point, I will just add this in to inicheck if you want for the next version. Mark has needed this too.

Reply:
Sounds good